app/frontend/actions.ts

```diff
@@ -550,7 +550,7 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
     let plan
     try {
       plan = await wallet.getTxPlan(args)
-      if (!plan.fee) {
+      if (!plan.fee && !plan.estimatedFee) {
```

Comment:
It seems to me, this would cause the `OutputTooSmall` error to never occur since if we cant build tx with big enough output, the function actually returns `estimatedFee`. 

I don't really see a reason to return the `estimatedFee` at all, since we don't use it anywhere. I am aware the currently implemented logic is not ideal since it was implemented in hurry few hours after the Shelley release and I am keeping it in mind for refactoring.

Reply:
I see, sorry I misunderstood the intent of the original code - I left a TODO comment to refactor this and I agree the behavior right now is counter-intuitive.

I also fixed the testcase for fee computation (`test/src/transaction-actions.js`) by mocking an utxo for the wallet generated from the test mnemonic so it is actually able to build a valid tx and estimate the fees which was the reason why I extended this condition in the first place thinking it is just an omission